### PR TITLE
EIP 658

### PR DESCRIPTION
**Pull request by adauto82**

### CODE REVIEW

File: rskj-core/src/main/java/org/ethereum/core/TransactionExecutor.java [hunk 1]
```diff
@@ -367,9 +386,27 @@ public void go() {
...
+        if (receipt == null) {
+            receipt = new TransactionReceipt();
+            long totalGasUsed = gasUsedInTheBlock + getGasUsed();
+            receipt.setCumulativeGas(totalGasUsed);
+            receipt.setTransaction(tx);
+            receipt.setLogInfoList(getVMLogs());
+            receipt.setGasUsed(getGasUsed());
+            receipt.setExecutionResult(getResult().getHReturn());
+            receipt.setError(executionError);
+//            receipt.setPostTxState(tr`ack.getRoot()); // TODO later when RepositoryTrack.getRoot() is implemented
```
Comment (ajlopez) [inline rskj-core/src/main/java/org/ethereum/core/TransactionExecutor.java]: Remove this commented code,, apparently it is not needed. The receipt.setPostTxState now has a different meaning.

And this getReceipt method is only used to create ANOTHER receipt? We should review the use cases for this method.

Why not set the receipt.setPostTxState according to, maybe, executionError content?

File: rskj-core/src/main/java/org/ethereum/core/TransactionReceipt.java [hunk 2]
```diff
@@ -196,11 +231,23 @@ public String toString() {
         // todo: fix that
 
         return "TransactionReceipt[" +
-                "\n  , postTxState=" + Hex.toHexString(postTxState) +
+                "\n  , " + (hasTxStatus() ? ("txStatus=" + (isTxStatusOK() ? "OK" : "FAILED"))
+                        : ("postTxState=" + Hex.toHexString(postTxState))) +
```
Comment (ajlopez) [inline rskj-core/src/main/java/org/ethereum/core/TransactionReceipt.java]: We could add the new fields, like the error field

File: rskj-core/src/main/java/org/ethereum/core/TransactionReceipt.java [hunk 3]
```diff
@@ -129,6 +145,8 @@ public Bloom getBloomFilter() {
         byte[] cumulativeGasRLP = RLP.encodeElement(this.cumulativeGas);
         byte[] gasUsedRLP = RLP.encodeElement(this.gasUsed);
         byte[] bloomRLP = RLP.encodeElement(this.bloomFilter.data);
+        byte[] result = RLP.encodeElement(this.executionResult);
+        byte[] err = RLP.encodeElement(this.error.getBytes(StandardCharsets.UTF_8));
```
Comment (SergioDemianLerner) [inline rskj-core/src/main/java/org/ethereum/core/TransactionReceipt.java]: I can't find where EIP 658 specifies that there should be an error message,

Comment (SergioDemianLerner) [reply rskj-core/src/main/java/org/ethereum/core/TransactionReceipt.java]: Note taken from another EIP:
[A recent EIP](https://github.com/ethereum/EIPs/pull/658) originally proposed adding return values to transaction receipts.  However, return data is not charged for (as it is not presently stored on the blockchain), so adding it to transaction receipts could result in DoS and spam opportunities. Instead, a simple Boolean `status` field was added to transaction receipts.  This was included in the Byzantium hard fork.

Comment (SergioDemianLerner) [reply rskj-core/src/main/java/org/ethereum/core/TransactionReceipt.java]: I really really really can't find any EIP or formal specification of this change in the Internet.  
It's not even here:
https://github.com/ethereum/go-ethereum/blob/master/core/types/receipt.go

**Comment by aeidelman** [comment]
@SergioDemianLerner is right, error message should not be included

File: rskj-core/src/main/java/org/ethereum/core/TransactionReceipt.java (hunk 3 above)
Comment (adauto82) [reply rskj-core/src/main/java/org/ethereum/core/TransactionReceipt.java]: I also could not find the spec that stated that we should use error. I just based my code on the PR of ethj. But we could find another way.

https://github.com/ethereum/ethereumj/blob/656301b4b146219b70b00ec4f9f8681583dd731c/ethereumj-core/src/main/java/org/ethereum/core/TransactionReceipt.java

**Comment by rskops** [comment]
SonarQube analysis reported 52 issues
* ![CRITICAL][CRITICAL] 6 critical
* ![MAJOR][MAJOR] 20 major
* ![MINOR][MINOR] 5 minor
* ![INFO][INFO] 21 info

#### Top 10 issues

1. ![CRITICAL][CRITICAL] [TransactionReceipt.java#L33](https://github.com/rsksmart/rskj/blob/33844250a77a85251338095e5691106364121b17/rskj-core/src/main/java/org/ethereum/core/TransactionReceipt.java#L33): Unused import - java.nio.charset.StandardCharsets. [rule](https://jenkins.rsklabs.io/sonar/coding_rules#rule_key=checkstyle%3Acom.puppycrawl.tools.checkstyle.checks.imports.UnusedImportsCheck)
1. ![CRITICAL][CRITICAL] [TransactionReceipt.java#L50](https://github.com/rsksmart/rskj/blob/33844250a77a85251338095e5691106364121b17/rskj-core/src/main/java/org/ethereum/core/TransactionReceipt.java#L50): Make this member "protected". [rule](https://jenkins.rsklabs.io/sonar/coding_rules#rule_key=squid%3AS2386)
1. ![CRITICAL][CRITICAL] [TransactionReceipt.java#L51](https://github.com/rsksmart/rskj/blob/33844250a77a85251338095e5691106364121b17/rskj-core/src/main/java/org/ethereum/core/TransactionReceipt.java#L51): Make this member "protected". [rule](https://jenkins.rsklabs.io/sonar/coding_rules#rule_key=squid%3AS2386)
1. ![CRITICAL][CRITICAL] [TransactionReceipt.java#L174](https://github.com/rsksmart/rskj/blob/33844250a77a85251338095e5691106364121b17/rskj-core/src/main/java/org/ethereum/core/TransactionReceipt.java#L174): Remove this call to "equals"; comparisons between unrelated arrays always return false. [rule](https://jenkins.rsklabs.io/sonar/coding_rules#rule_key=squid%3AS2159)
1. ![CRITICAL][CRITICAL] [TransactionReceipt.java#L176](https://github.com/rsksmart/rskj/blob/33844250a77a85251338095e5691106364121b17/rskj-core/src/main/java/org/ethereum/core/TransactionReceipt.java#L176): Remove this call to "equals"; comparisons between unrelated arrays always return false. [rule](https://jenkins.rsklabs.io/sonar/coding_rules#rule_key=squid%3AS2159)
1. ![CRITICAL][CRITICAL] [TransactionReceipt.java#L182](https://github.com/rsksmart/rskj/blob/33844250a77a85251338095e5691106364121b17/rskj-core/src/main/java/org/ethereum/core/TransactionReceipt.java#L182): Remove this call to "equals"; comparisons between unrelated arrays always return false. [rule](https://jenkins.rsklabs.io/sonar/coding_rules#rule_key=squid%3AS2159)
1. ![MAJOR][MAJOR] [BlockExecutor.java#L152](https://github.com/rsksmart/rskj/blob/33844250a77a85251338095e5691106364121b17/rskj-core/src/main/java/co/rsk/core/bc/BlockExecutor.java#L152): Method co.rsk.core.bc.BlockExecutor.validate(Block, BlockResult) appears to call the same method on the same object redundantly [rule](https://jenkins.rsklabs.io/sonar/coding_rules#rule_key=fb-contrib%3APRMC_POSSIBLY_REDUNDANT_METHOD_CALLS)
1. ![MAJOR][MAJOR] [BlockExecutor.java#L229](https://github.com/rsksmart/rskj/blob/33844250a77a85251338095e5691106364121b17/rskj-core/src/main/java/co/rsk/core/bc/BlockExecutor.java#L229): Method co.rsk.core.bc.BlockExecutor.execute(Block, byte[], boolean, boolean) does not presize the allocation of a collection [rule](https://jenkins.rsklabs.io/sonar/coding_rules#rule_key=fb-contrib%3APSC_PRESIZE_COLLECTIONS)
1. ![MAJOR][MAJOR] [BlockExecutor.java#L263](https://github.com/rsksmart/rskj/blob/33844250a77a85251338095e5691106364121b17/rskj-core/src/main/java/co/rsk/core/bc/BlockExecutor.java#L263): Method co.rsk.core.bc.BlockExecutor.execute(Block, byte[], boolean, boolean) does not presize the allocation of a collection [rule](https://jenkins.rsklabs.io/sonar/coding_rules#rule_key=fb-contrib%3APSC_PRESIZE_COLLECTIONS)
1. ![MAJOR][MAJOR] [TransactionExecutor.java#L95](https://github.com/rsksmart/rskj/blob/33844250a77a85251338095e5691106364121b17/rskj-core/src/main/java/org/ethereum/core/TransactionExecutor.java#L95): Constructor has 8 parameters, which is greater than 7 authorized. [rule](https://jenkins.rsklabs.io/sonar/coding_rules#rule_key=squid%3AS00107)

[CRITICAL]: https://sonarsource.github.io/sonar-github/severity-critical.png 'Severity: CRITICAL'
[INFO]: https://sonarsource.github.io/sonar-github/severity-info.png 'Severity: INFO'
[MAJOR]: https://sonarsource.github.io/sonar-github/severity-major.png 'Severity: MAJOR'
[MINOR]: https://sonarsource.github.io/sonar-github/severity-minor.png 'Severity: MINOR'

**Comment by rskops** [comment]
SonarQube analysis reported 51 issues
* ![CRITICAL][CRITICAL] 6 critical
* ![MAJOR][MAJOR] 19 major
* ![MINOR][MINOR] 5 minor
* ![INFO][INFO] 21 info

#### Top 10 issues

1. ![CRITICAL][CRITICAL] [TransactionReceipt.java#L33](https://github.com/rsksmart/rskj/blob/33844250a77a85251338095e5691106364121b17/rskj-core/src/main/java/org/ethereum/core/TransactionReceipt.java#L33): Unused import - java.nio.charset.StandardCharsets. [rule](https://jenkins.rsklabs.io/sonar/coding_rules#rule_key=checkstyle%3Acom.puppycrawl.tools.checkstyle.checks.imports.UnusedImportsCheck)
1. ![CRITICAL][CRITICAL] [TransactionReceipt.java#L50](https://github.com/rsksmart/rskj/blob/33844250a77a85251338095e5691106364121b17/rskj-core/src/main/java/org/ethereum/core/TransactionReceipt.java#L50): Make this member "protected". [rule](https://jenkins.rsklabs.io/sonar/coding_rules#rule_key=squid%3AS2386)
1. ![CRITICAL][CRITICAL] [TransactionReceipt.java#L51](https://github.com/rsksmart/rskj/blob/33844250a77a85251338095e5691106364121b17/rskj-core/src/main/java/org/ethereum/core/TransactionReceipt.java#L51): Make this member "protected". [rule](https://jenkins.rsklabs.io/sonar/coding_rules#rule_key=squid%3AS2386)
1. ![CRITICAL][CRITICAL] [TransactionReceipt.java#L173](https://github.com/rsksmart/rskj/blob/33844250a77a85251338095e5691106364121b17/rskj-core/src/main/java/org/ethereum/core/TransactionReceipt.java#L173): Remove this call to "equals"; comparisons between unrelated arrays always return false. [rule](https://jenkins.rsklabs.io/sonar/coding_rules#rule_key=squid%3AS2159)
1. ![CRITICAL][CRITICAL] [TransactionReceipt.java#L175](https://github.com/rsksmart/rskj/blob/33844250a77a85251338095e5691106364121b17/rskj-core/src/main/java/org/ethereum/core/TransactionReceipt.java#L175): Remove this call to "equals"; comparisons between unrelated arrays always return false. [rule](https://jenkins.rsklabs.io/sonar/coding_rules#rule_key=squid%3AS2159)
1. ![CRITICAL][CRITICAL] [TransactionReceipt.java#L181](https://github.com/rsksmart/rskj/blob/33844250a77a85251338095e5691106364121b17/rskj-core/src/main/java/org/ethereum/core/TransactionReceipt.java#L181): Remove this call to "equals"; comparisons between unrelated arrays always return false. [rule](https://jenkins.rsklabs.io/sonar/coding_rules#rule_key=squid%3AS2159)
1. ![MAJOR][MAJOR] [BlockExecutor.java#L152](https://github.com/rsksmart/rskj/blob/33844250a77a85251338095e5691106364121b17/rskj-core/src/main/java/co/rsk/core/bc/BlockExecutor.java#L152): Method co.rsk.core.bc.BlockExecutor.validate(Block, BlockResult) appears to call the same method on the same object redundantly [rule](https://jenkins.rsklabs.io/sonar/coding_rules#rule_key=fb-contrib%3APRMC_POSSIBLY_REDUNDANT_METHOD_CALLS)
1. ![MAJOR][MAJOR] [BlockExecutor.java#L229](https://github.com/rsksmart/rskj/blob/33844250a77a85251338095e5691106364121b17/rskj-core/src/main/java/co/rsk/core/bc/BlockExecutor.java#L229): Method co.rsk.core.bc.BlockExecutor.execute(Block, byte[], boolean, boolean) does not presize the allocation of a collection [rule](https://jenkins.rsklabs.io/sonar/coding_rules#rule_key=fb-contrib%3APSC_PRESIZE_COLLECTIONS)
1. ![MAJOR][MAJOR] [BlockExecutor.java#L263](https://github.com/rsksmart/rskj/blob/33844250a77a85251338095e5691106364121b17/rskj-core/src/main/java/co/rsk/core/bc/BlockExecutor.java#L263): Method co.rsk.core.bc.BlockExecutor.execute(Block, byte[], boolean, boolean) does not presize the allocation of a collection [rule](https://jenkins.rsklabs.io/sonar/coding_rules#rule_key=fb-contrib%3APSC_PRESIZE_COLLECTIONS)
1. ![MAJOR][MAJOR] [TransactionExecutor.java#L94](https://github.com/rsksmart/rskj/blob/33844250a77a85251338095e5691106364121b17/rskj-core/src/main/java/org/ethereum/core/TransactionExecutor.java#L94): Constructor has 8 parameters, which is greater than 7 authorized. [rule](https://jenkins.rsklabs.io/sonar/coding_rules#rule_key=squid%3AS00107)

[CRITICAL]: https://sonarsource.github.io/sonar-github/severity-critical.png 'Severity: CRITICAL'
[INFO]: https://sonarsource.github.io/sonar-github/severity-info.png 'Severity: INFO'
[MAJOR]: https://sonarsource.github.io/sonar-github/severity-major.png 'Severity: MAJOR'
[MINOR]: https://sonarsource.github.io/sonar-github/severity-minor.png 'Severity: MINOR'

**Comment by rskops** [comment]
SonarQube analysis reported 41 issues
* ![MAJOR][MAJOR] 16 major
* ![MINOR][MINOR] 4 minor
* ![INFO][INFO] 21 info

#### Top 10 issues

1. ![MAJOR][MAJOR] [BlockExecutor.java#L152](https://github.com/rsksmart/rskj/blob/699a5bee6e815da00b74a7d5683f5e3d2ec69065/rskj-core/src/main/java/co/rsk/core/bc/BlockExecutor.java#L152): Method co.rsk.core.bc.BlockExecutor.validate(Block, BlockResult) appears to call the same method on the same object redundantly [rule](https://jenkins.rsklabs.io/sonar/coding_rules#rule_key=fb-contrib%3APRMC_POSSIBLY_REDUNDANT_METHOD_CALLS)
1. ![MAJOR][MAJOR] [BlockExecutor.java#L229](https://github.com/rsksmart/rskj/blob/699a5bee6e815da00b74a7d5683f5e3d2ec69065/rskj-core/src/main/java/co/rsk/core/bc/BlockExecutor.java#L229): Method co.rsk.core.bc.BlockExecutor.execute(Block, byte[], boolean, boolean) does not presize the allocation of a collection [rule](https://jenkins.rsklabs.io/sonar/coding_rules#rule_key=fb-contrib%3APSC_PRESIZE_COLLECTIONS)
1. ![MAJOR][MAJOR] [BlockExecutor.java#L263](https://github.com/rsksmart/rskj/blob/699a5bee6e815da00b74a7d5683f5e3d2ec69065/rskj-core/src/main/java/co/rsk/core/bc/BlockExecutor.java#L263): Method co.rsk.core.bc.BlockExecutor.execute(Block, byte[], boolean, boolean) does not presize the allocation of a collection [rule](https://jenkins.rsklabs.io/sonar/coding_rules#rule_key=fb-contrib%3APSC_PRESIZE_COLLECTIONS)
1. ![MAJOR][MAJOR] [TransactionExecutor.java#L94](https://github.com/rsksmart/rskj/blob/699a5bee6e815da00b74a7d5683f5e3d2ec69065/rskj-core/src/main/java/org/ethereum/core/TransactionExecutor.java#L94): Constructor has 8 parameters, which is greater than 7 authorized. [rule](https://jenkins.rsklabs.io/sonar/coding_rules#rule_key=squid%3AS00107)
1. ![MAJOR][MAJOR] [TransactionExecutor.java#L99](https://github.com/rsksmart/rskj/blob/699a5bee6e815da00b74a7d5683f5e3d2ec69065/rskj-core/src/main/java/org/ethereum/core/TransactionExecutor.java#L99): Constructor has 10 parameters, which is greater than 7 authorized. [rule](https://jenkins.rsklabs.io/sonar/coding_rules#rule_key=squid%3AS00107)
1. ![MAJOR][MAJOR] [TransactionExecutor.java#L122](https://github.com/rsksmart/rskj/blob/699a5bee6e815da00b74a7d5683f5e3d2ec69065/rskj-core/src/main/java/org/ethereum/core/TransactionExecutor.java#L122): The Cyclomatic Complexity of this method "init" is 14 which is greater than 10 authorized. [rule](https://jenkins.rsklabs.io/sonar/coding_rules#rule_key=squid%3AMethodCyclomaticComplexity)
1. ![MAJOR][MAJOR] [TransactionExecutor.java#L141](https://github.com/rsksmart/rskj/blob/699a5bee6e815da00b74a7d5683f5e3d2ec69065/rskj-core/src/main/java/org/ethereum/core/TransactionExecutor.java#L141): Method org.ethereum.core.TransactionExecutor.init() appears to call the same method on the same object redundantly [rule](https://jenkins.rsklabs.io/sonar/coding_rules#rule_key=fb-contrib%3APRMC_POSSIBLY_REDUNDANT_METHOD_CALLS)
1. ![MAJOR][MAJOR] [TransactionExecutor.java#L142](https://github.com/rsksmart/rskj/blob/699a5bee6e815da00b74a7d5683f5e3d2ec69065/rskj-core/src/main/java/org/ethereum/core/TransactionExecutor.java#L142): Method org.ethereum.core.TransactionExecutor.init() appears to call the same method on the same object redundantly [rule](https://jenkins.rsklabs.io/sonar/coding_rules#rule_key=fb-contrib%3APRMC_POSSIBLY_REDUNDANT_METHOD_CALLS)
1. ![MAJOR][MAJOR] [TransactionExecutor.java#L280](https://github.com/rsksmart/rskj/blob/699a5bee6e815da00b74a7d5683f5e3d2ec69065/rskj-core/src/main/java/org/ethereum/core/TransactionExecutor.java#L280): Take the required action to fix the issue indicated by this comment. [rule](https://jenkins.rsklabs.io/sonar/coding_rules#rule_key=squid%3AS1134)
1. ![MAJOR][MAJOR] [TransactionExecutor.java#L338](https://github.com/rsksmart/rskj/blob/699a5bee6e815da00b74a7d5683f5e3d2ec69065/rskj-core/src/main/java/org/ethereum/core/TransactionExecutor.java#L338): The Cyclomatic Complexity of this method "go" is 11 which is greater than 10 authorized. [rule](https://jenkins.rsklabs.io/sonar/coding_rules#rule_key=squid%3AMethodCyclomaticComplexity)

[INFO]: https://sonarsource.github.io/sonar-github/severity-info.png 'Severity: INFO'
[MAJOR]: https://sonarsource.github.io/sonar-github/severity-major.png 'Severity: MAJOR'
[MINOR]: https://sonarsource.github.io/sonar-github/severity-minor.png 'Severity: MINOR'

**Pull request merged**